process/indexer/indexer.go

```diff
@@ -1,35 +1,46 @@
 package indexer
 
 import (
+	"github.com/multiversx/mx-chain-core-go/core/check"
 	"github.com/multiversx/mx-chain-core-go/data/outport"
 	"github.com/multiversx/mx-chain-sovereign-notifier-go/process"
 )
 
 type indexer struct {
 	notifier process.SovereignNotifier
+	cache    LRUOutportBlockCache
 }
 
 // NewIndexer creates an indexer which will internally save received blocks
 // and notify sovereign shards for each finalized block
-func NewIndexer(notifier process.SovereignNotifier) (process.Indexer, error) {
-	if notifier == nil {
+func NewIndexer(notifier process.SovereignNotifier, cache LRUOutportBlockCache) (process.Indexer, error) {
+	if check.IfNil(notifier) {
 		return nil, errNilSovereignNotifier
 	}
+	if check.IfNil(cache) {
+		return nil, errNilOutportBlockCache
+	}
 
 	return &indexer{
+		cache:    cache,
 		notifier: notifier,
 	}, nil
 }
 
 // SaveBlock will save the received block in an internal cache
-func (i *indexer) SaveBlock(_ *outport.OutportBlock) error {
-	return nil
+func (i *indexer) SaveBlock(outportBlock *outport.OutportBlock) error {
+	return i.cache.Add(outportBlock)
```

Comment:
is this a good approach to using an LRU cache? What happens in the following scenario:
We call this function 1000 times because the shard got stuck and stopped notarizing metachain for an arbitrary reason. The call to FinalizedBlock will come, maybe, after another 1000 calls to SaveBlock. The cache won't be able to retain all data and will evict old elements resulting in data loss.
Strongly suggest renaming the interface to something that does not contain the LRU keyword and the implementation should be done with a map. No maximum size is needed for that component.

Reply:
Very good catch and suggestion.
Refactored to use `Extract`